solutions/cf-1216C.mdx

```diff
@@ -0,0 +1,240 @@
+  def __init__(self, a, b, c, d):
+    self.x1,self.y1,self.x2,self.y2 = a, b, c, d
+  def area(self):
+    return (self.x2-self.x1)*(self.y2-self.y1)
+
+def cut(a, b):
+  if(a.x1>=b.x1 and b.x2>=a.x1 and b.y1<=a.y1 and b.y2>=a.y2):
+		#if B cuts A from the left side
+	  a.x1 = b.x2
+	  a.x2 = max(a.x2, b.x2) # if b covers A entirely
+  if(a.x2>=b.x1 and b.x2>=a.x2 and b.y1<=a.y1 and b.y2>=a.y2):
+	  #if B cuts A from the right side
+	  a.x2 = b.x1;
+	  a.x1 = min(a.x1, b.x1) # if b covers A entirely
+  if(a.y1>=b.y1 and b.y2>=a.y1 and b.x1<=a.x1 and b.x2>=a.x2):
+		#if B cuts A from the bottom side
+	  a.y1 = b.y2;
+	  a.y2 = max(a.y2, b.y2) # if b covers A entirely
+  if(a.y2>=b.y1 and b.y2>=a.y2 and b.x1<=a.x1 and b.x2>=a.x2):
+    #if B cuts A from the top side
+    a.y2 = b.y1;
+    a.y1 = min(a.y1, b.y1) #if b covers A entirely
+	
+  return a
+x = input().split()
+y = input().split()
+z = input().split()
+
+a = Rect(int(x[0]), int(x[1]), int(x[2]), int(x[3]))
+b = Rect(int(y[0]), int(y[1]), int(y[2]), int(y[3]))
+c = Rect(int(z[0]), int(z[1]), int(z[2]), int(z[3]))
+
+a = cut(a, b);
+a = cut(a, c);
+if(a.x2-a.x1==0 || a.y2-a.y1==0): #area!=0
+  print("YES")
+else:
+  print("NO")
+```
+</PySection>
+
+<JavaSection>
+```java
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+public class Main {
+  
+  public static void main(String[] args) throws IOException {
+    Kattio io = new Kattio();
+		Rect a = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    Rect b = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    Rect c = new Rect(io.nextInt(), io.nextInt(), io.nextInt(), io.nextInt());
+    a = cut(a, b);
+  	a = cut(a, c);
+  	if(a.x2-a.x1==0 || a.y2-a.y1==0) //area!=0
+  	System.out.println("NO");
+  	else
+  	System.out.println("YES");
+    io.close();
+  }
+  static class Kattio extends PrintWriter {
+		private BufferedReader r;
+		private StringTokenizer st;
+		// standard input
+		public Kattio() { this(System.in, System.out); }
+		public Kattio(InputStream i, OutputStream o) {
+			super(o);
+			r = new BufferedReader(new InputStreamReader(i));
+		}
+		// USACO-style file input
+		public Kattio(String problemName) throws IOException {
+			super(problemName + ".out");
+			r = new BufferedReader(new FileReader(problemName + ".in"));
+		}
+		// returns null if no more input
+		public String next() {
+			try {
+				while (st == null || !st.hasMoreTokens())
+					st = new StringTokenizer(r.readLine());
+				return st.nextToken();
+			} catch (Exception e) { }
+			return null;
+		}
+		public int nextInt() { return Integer.parseInt(next()); }
+		public double nextDouble() { return Double.parseDouble(next()); }
+		public long nextLong() { return Long.parseLong(next()); }
+	}
+  static class Rect{
+    int x1,y1,x2,y2;
+    public Rect(int a, int b, int c, int d){
+      x1 = a;
+      y1 = b;
+      x2 = c;
+      y2 = d;
+    }
+  	int area()
+  	{
+  		return (x2-x1)*(y2-y1);
+  	}
+  }
```

Comment:
something about the brackets is off putting they aren't aligned with the code so fix it